qa/tasks/cephfs/xfstests_dev.py

```diff
@@ -0,0 +1,179 @@
+        from teuthology.misc import get_system_type
+
+        args = ['sudo', 'install', '-y']
+
+        distro = get_system_type(self.mount_a.client_remote,
+                                 distro=True).lower()
+        if distro in ['redhatenterpriseserver', 'fedora', 'centos']:
+            deps = """acl attr automake bc dbench dump e2fsprogs fio \
+            gawk gcc indent libtool lvm2 make psmisc quota sed \
+            xfsdump xfsprogs \
+            libacl-devel libattr-devel libaio-devel libuuid-devel \
+            xfsprogs-devel btrfs-progs-devel python sqlite""".split()
+            deps_for_old_distros = "xfsprogs-qa-devel"
+
+            args.insert(1, 'yum')
+            args.extend(deps)
+            args.append(deps_for_old_distros)
+        elif distro == 'ubuntu':
+            deps = """xfslibs-dev uuid-dev libtool-bin \
+            e2fsprogs automake gcc libuuid1 quota attr libattr1-dev make \
+            libacl1-dev libaio-dev xfsprogs libgdbm-dev gawk fio dbench \
+            uuid-runtime python sqlite3""".split()
+
+            args.insert(1, 'apt-get')
+            args.extend(deps)
+        else:
+            raise RuntimeError('expected a yum based or a apt based system')
+
+        proc = self.mount_a.client_remote.run(args=args, omit_sudo=False)
+
+    def create_reqd_users(self):
+        self.mount_a.client_remote.run(args=['sudo', 'useradd', 'fsgqa'],
+                                       omit_sudo=False, check_status=False)
+        self.mount_a.client_remote.run(args=['sudo', 'groupadd', 'fsgqa'],
+                                       omit_sudo=False, check_status=False)
+        self.mount_a.client_remote.run(args=['sudo', 'useradd',
+                                             '123456-fsgqa'], omit_sudo=False,
+                                       check_status=False)
+
+    def write_local_config(self):
+        from os.path import join
+        from textwrap import dedent
+        from teuthology.misc import sudo_write_file
+
+        def get_a_mon_socket_addr():
+            from json import loads
+
+            mon_status = loads(self.fs.mon_manager.raw_cluster_cmd(
+                'mon_status'))
+            mon_socket_addr = mon_status['monmap']['mons'][0]['public_addrs']\
+                ['addrvec'][1]['addr']
+            return mon_socket_addr
+
+        mon_socket_addr = get_a_mon_socket_addr()
+        self.test_dev = mon_socket_addr + ':/' + self.test_dirname
+        self.scratch_dev = mon_socket_addr + ':/' + self.scratch_dirname
+
+        xfstests_config_contents = dedent('''\
+            export FSTYP=ceph
+            export TEST_DEV={}
+            export TEST_DIR={}
+            #export SCRATCH_DEV={}
+            #export SCRATCH_MNT={}
+            export TEST_FS_MOUNT_OPTS="-o name=admin,secret={}"
+            ''').format(self.test_dev, self.test_dirs_mount_path, self.scratch_dev,
+                        self.scratch_dirs_mount_path, self.get_admin_key())
```

Comment:
Presumably `TEST_FS_MOUNT_OPS` is so xfs-tests can remount?

The monitor addresses should no longer be necessary now that mount.ceph gets the information from ceph.conf. See `kernel_mount.py` and recent changes made by Jeff.